crates/re_log_types/src/component_types/mesh3d.rs

```diff
@@ -137,31 +137,31 @@ pub enum RawMeshError {
 /// );
 /// ```
 #[derive(ArrowField, ArrowSerialize, ArrowDeserialize, Clone, Debug, PartialEq)]
-#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
 pub struct RawMesh3D {
     pub mesh_id: MeshId,
 
     /// The flattened vertex positions array of this mesh.
     ///
     /// The length of this vector should always be divisible by three (since this is a 3D mesh).
     ///
-    /// If no indices are specified, then each triplet of vertex positions are intrpreted as a triangle
+    /// If no indices are specified, then each triplet of vertex positions are interpreted as a triangle
     /// and the length of this must be divisible by 9.
-    pub vertex_positions: Vec<f32>,
+    pub vertex_positions: Buffer<f32>,
 
     /// Per-vertex albedo colors.
-    pub vertex_colors: Option<Vec<ColorRGBA>>,
+    /// This is actually an encoded [`super::ColorRGBA`]
+    pub vertex_colors: Option<Buffer<u32>>,
```

Comment:
Hum, why doesn't `Buffer<ColorRGBA>` work?

Reply:
In short because Buffer is implemented as:

```
// Blanket implementation for Buffer
impl<T> ArrowDeserialize for Buffer<T>
where
    T: ArrowDeserialize + NativeType + ArrowEnableVecForType,
    for<'b> &'b <T as ArrowDeserialize>::ArrayType: IntoIterator,
{
    type ArrayType = ListArray<i32>;

    #[inline]
    fn arrow_deserialize(
        v: <&Self::ArrayType as IntoIterator>::Item,
    ) -> Option<<Self as ArrowField>::Type> {
        v.map(|t| {
            t.as_any()
                .downcast_ref::<PrimitiveArray<T>>()
                .unwrap()
                .values()
                .clone()
        })
    }
}
```

Similar to our workaround for BinaryBuffer, we could probably do something like implement a `struct ColorRGBABuffer(Buffer<ColorRGBA>)`, to specialize it.

I was hoping I could just make ColorRGBA act as a native type, but arrow2::NativeType is sealed: https://github.com/jorgecarleitao/arrow2/blob/main/src/types/native.rs